### PR TITLE
[2.7][Validator] Added checkDNS option on URL validator

### PR DESCRIPTION
**Pull request by saro0h**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #10088 |
| License | MIT |
| Doc PR | symfony/symfony-docs#4631 |

### CODE REVIEW

**Comment by hhamon** [comment]
Looks good!

File: src/Symfony/Component/Validator/Constraints/UrlValidator.php [hunk 1]
```diff
@@ -62,6 +62,16 @@ public function validate($value, Constraint $constraint)
             $this->buildViolation($constraint->message)
                 ->setParameter('{{ value }}', $this->formatValue($value))
                 ->addViolation();
+
+            return;
+        }
+
+        $host = parse_url($value, PHP_URL_HOST);
```
Comment (Tobion) [inline src/Symfony/Component/Validator/Constraints/UrlValidator.php]: no need to call parse_url when checkDNS is false

**Comment by saro0h** [comment]
Changes have been made.

File: src/Symfony/Component/Validator/Constraints/UrlValidator.php [hunk 2]
```diff
@@ -62,6 +62,18 @@ public function validate($value, Constraint $constraint)
...
                 ->addViolation();
+
+            return;
+        }
+
+        if ($constraint->checkDNS) {
+            $host = parse_url($value, PHP_URL_HOST);
+
+            if (!checkdnsrr($host, 'ANY')) {
+                $this->buildViolation($constraint->message)
```
Comment (fabpot) [inline src/Symfony/Component/Validator/Constraints/UrlValidator.php]: The message in this case should be different as the URL is valid. It just does not resolve to anything.

**Comment by saro0h** [comment]
Changes have been made.

**Comment by fabpot** [comment]
:+1:

File: src/Symfony/Component/Validator/Constraints/Url.php [hunk 3]
```diff
@@ -24,5 +24,7 @@
 class Url extends Constraint
 {
     public $message = 'This value is not a valid URL.';
+    public $dnsMessage = 'The host could not be resolved.';
```
Comment (javiereguiluz) [inline src/Symfony/Component/Validator/Constraints/Url.php]: Should this new message be added to `src/Symfony/Component/Validator/Resources/translations/validators.en.xlf` file in order to translate it?

Comment (saro0h) [inline src/Symfony/Component/Validator/Constraints/Url.php]: Done, thank you!

**Comment by fabpot** [comment]
Thank you @saro0h.

**Pull request merged**